components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/status_tray/l10n_util.cc

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/status_tray/l10n_util.h"
+
+#include <string>
+#include <vector>
+
+#include "base/no_destructor.h"
+#include "base/notreached.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/win/atl.h"
+#include "base/win/embedded_i18n/language_selector.h"
+#include "base/win/i18n.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/status_tray/resources/wireguard_strings.h"
+
+namespace brave_vpn {
+namespace {
+
+constexpr base::win::i18n::LanguageSelector::LangToOffset
+    kLanguageOffsetPairs[] = {
+#define HANDLE_LANGUAGE(l_, o_) {L## #l_, o_},
+        DO_LANGUAGES
+#undef HANDLE_LANGUAGE
+};
+
+std::wstring GetPreferredLanguage() {
+  std::vector<std::wstring> languages;
+  if (!base::win::i18n::GetUserPreferredUILanguageList(&languages) ||
+      languages.size() == 0) {
+    return L"en-us";
+  }
+
+  return languages[0];
+}
+
+const base::win::i18n::LanguageSelector& GetLanguageSelector() {
+  static base::NoDestructor<base::win::i18n::LanguageSelector> instance(
+      GetPreferredLanguage(), kLanguageOffsetPairs);
+  return *instance;
+}
+
+}  // namespace
+
+std::u16string GetLocalizedString(UINT base_message_id) {
```

Comment:
I don't understand what's going on here, the chromium code already takes care of this. This should be added to a pak file just like in other cases and use `ui::ResourceBundle::GetSharedInstance()`

Reply:
the point is that this executable is installed with Brave so it can just load the pak files that come with Brave using `ui::ResourceBundle`

Reply:
This is different from the installer where Brave may or may not already exist when it runs

Reply:
I tried to to this and it doesnt work, the resources seems stripped.
```
[0707/222952.392:FATAL:resource_bundle.cc(1199)] Check failed: !data.empty(). Unable to find resource: 54404. If this happens in a browser test running on Windows, it may be that dead-code elimination stripped out the code that uses the resource, causing the resource to be stripped out because the resource is not used by chrome.dll. See https://crbug.com/1181150.
```
Also I do not see similar usages of `InitSharedInstanceWithPakPath`, it is used mostly in tests or to load test paks from examples ([link](https://source.chromium.org/search?q=InitSharedInstanceWithPakPath&ss=chromium))

Opened separate PR for this approach to play if need https://github.com/brave/brave-core/pull/19202/files, it fails in runtime because no strings found in the pak file. 

Additionally worth to mention that using runtime pak files we implicitly dependent on external resource. Explicit better than implicit and Chromium recommended way to do compile time localization for separate executables is using `generate_embedded_i18n` which `defines a rule to generate localized string .rc and .h files that can be embedded directly into a library or executable if desired` ([link](https://source.chromium.org/chromium/chromium/src/+/main:base/win/embedded_i18n/generate_embedded_i18n.gni;l=8?q=generate_embedded_i18n&ss=chromium)) and used in others Chromium executables ([credential_provider](https://source.chromium.org/chromium/chromium/src/+/main:chrome/credential_provider/gaiacp/gcp_utils.cc;l=923?q=gcp_utils.cc&ss=chromium%2Fchromium%2Fsrc:chrome%2Fcredential_provider%2F), [updater](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:chrome/updater/win/ui/l10n_util.cc;l=47), [installer](https://source.chromium.org/chromium/chromium/src/+/main:chrome/installer/util/l10n_string_util.cc;l=66?q=GetLocalizedString&ss=chromium)).